var/spack/repos/builtin/packages/quda/package.py

```diff
@@ -0,0 +1,192 @@
+    variant("bqcd", default=False, description="Enable BQCD interface")
+    variant("cps", default=False, description="Enable CPS interface")
+    variant("qdpjit", default=False, description="Enable QDPJIT interface")
+    variant("tifr", default=False, description="Enable TIFR interface")
+    variant("multigrid", default=False, description="Enable multigrid")
+    variant("nvshmem", default=False, description="Enable NVSHMEM", when="+cuda")
+
+    variant("clover", default=False, description="Build clover Dirac operators")
+    variant(
+        "clover_hasenbusch", default=False, description="Build clover Hasenbusch twist operators"
+    )
+    variant("domain_wall", default=False, description="Build domain wall Dirac operators")
+    variant("laplace", default=False, description="Build laplace operator")
+    variant(
+        "ndeg_twisted_clover",
+        default=False,
+        description="Build non-degenerate twisted clover Dirac operators",
+    )
+    variant(
+        "ndeg_twisted_mass",
+        default=False,
+        description="Build non-degenerate twisted mass Dirac operators",
+    )
+    variant("staggered", default=False, description="Build staggered Dirac operators")
+    variant("twisted_clover", default=False, description="Build twisted clover Dirac operators")
+    variant("twisted_mass", default=False, description="Build twisted mass Dirac operators")
+    variant("wilson", default=True, description="Build Wilson Dirac operators")
+
+    with when("+multigrid"):
+        variant(
+            "mg_mrhs_list",
+            default=16,
+            multi=True,
+            description="The list of multi-rhs sizes that get compiled",
+        )
+        variant(
+            "mg_nvec_list",
+            default="6,24,32",
+            multi=True,
+            description="The list of null space vector sizes that get compiled",
+        )
+
+    # dependencies
+    depends_on("mpi", when="+mpi")
+    depends_on("cuda", when="+cuda")
+    depends_on("nvshmem", when="+nvshmem")
+    depends_on("gdrcopy", when="+nvshmem")
+    with when("+rocm"):
+        depends_on("hip")
+        depends_on("hipblas")
+        depends_on("hipfft")
+        depends_on("hiprand")
+        depends_on("hipcub")
+
+    conflicts("+qmp +mpi", msg="Specifying both QMP and MPI might result in undefined behavior")
+    conflicts("+cuda +rocm", msg="CUDA and ROCm support are mutually exclusive")
+    conflicts("~cuda ~rocm", msg="Either CUDA or ROCm support is required")
+    conflicts("cuda_arch=none", when="+cuda", msg="Please indicate a cuda_arch value")
+    conflicts("+nvshmem", when="~mpi ~qmp", msg="NVSHMEM requires either +mpi or +qmp to be enabled")
+
+    # CMAKE_BUILD_TYPE
+    # No support for RelWithDebInfo and MinSizeRel
+    variant(
+        "build_type",
+        default="STRICT",
+        description="The build type to build",
+        values=("STRICT", "RELEASE", "Release", "DEVEL", "DEBUG", "Debug", "HOSTDEBUG", "SANITIZE")
```

Comment:
FYI, https://cmake.org/cmake/help/latest/manual/cmake-buildsystem.7.html#case-sensitivity

It is possible that `RELEASE` and `Release` are going to map to the same (same for all uppercase). Since spack typically uses CamelCase, it may be worth testing it `Strict`, `Devel`, `HostDebug`, and `Sanitize` just work. There would be no need to duplicate `DEBUG` and `Debug`, same with `RELEASE` and `Release`.

Reply:
Let me know what you would prefer.

Reply:
I checked CamelCase versions. They don't work unless put as values `values=(...)`. Then I would prefer the build types fully capitalized as they are in the `CMakeLists.txt` as
```suggestion
        values=("STRICT", "RELEASE",  "DEVEL", "DEBUG", "HOSTDEBUG", "SANITIZE")
```

Reply:
Ok, sounds good. Go ahead and commit. I'll review, approve, merge.